include/tvm/ir/transform.h

```diff
@@ -414,6 +414,65 @@ TVM_DLL Pass
 CreateModulePass(const runtime::TypedPackedFunc<IRModule(IRModule, PassContext)>& pass_func,
                  int opt_level, String name, Array<runtime::String> required);
 
+class ModulePassNode : public PassNode {
```

Comment:
I wonder why we need to move those pass definitions to a different place..Is this related to the error messages?

Reply:
It is needed for `CreateModulePass` (https://github.com/apache/tvm/pull/7152/files#diff-99b9cb940bf19f00bb0d5956a156574f1f1940b026bff03fc0cb5f78c71d9faeR471). The new version of `CreateModulePass` needs to be templated so it accepts lambdas (and other functions) before they have been converted to `TypedPackedFunc`s. That way we can correctly set the name.

Reply:
I do think it is possible to keep things as they are(not moving to the header). Since the original rationale is to bring benefit of isolation. Ideally we don't want to cut one part to make things easier in another case. 

To resolve this, one way is to keep the API in their original form, while creating a separate function CreateModulePass_ that takes the `runtime::TypedPackedFunc` which constructs during the templated version. 



One possible way is to create an overloaded function of CreateModulePass_ that accepts the runtime::TypedPackedFunc which get calls into it.

Reply:
Additionally, this could be something that we can add back once the packed func land

Reply:
We cannot take a `TypedPackedFunc` in `CreateModulePass` because the name needs to be provided to the constructor of `TypedPackedFunc`. We cannot set it after the fact.

Reply:
To clarify. 
- Have a function (not templated) `CreateModulePass_` that takes in `TypedPackedFunc`
- Create a templated function `CreateModulePass` that constructs the function(with name) and call into `CreateModulePass_`